packages/react/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,162 @@
+import { ReactNode, useEffect, useState } from 'react';
+
+import { AvatarProps } from './Avatar.types';
+
+import clsx from 'clsx';
+import { avatarClasses, getAvatarUtilityClass } from './Avatar.classes';
+
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+
+import { styled, useThemeProps } from '@mui/material/styles';
+import { capitalize } from '@mui/material';
+
+import { svgIconClasses } from '../SvgIcon';
+
+type AvatarOwnerState = {
+  classes?: AvatarProps['classes'];
+  size: NonNullable<AvatarProps['size']>;
+  variant: NonNullable<AvatarProps['variant']>;
+};
+
+const useUtilityClasses = (ownerState: AvatarOwnerState) => {
+  const { classes, variant = 'square' } = ownerState;
+
+  const slots = {
+    root: ['root', `variant${capitalize(variant)}`],
+    image: ['image']
+  };
+
+  return composeClasses(slots, getAvatarUtilityClass, classes);
+};
+
+function useLoaded(src: string): 'loaded' | 'error' | null {
+  const [loaded, setLoaded] = useState<'loaded' | 'error' | null>(null);
+
+  useEffect(() => {
+    if (!src) {
+      return undefined;
+    }
+
+    setLoaded(null);
+
+    let active = true;
+    const image = new Image();
+    image.onload = () => {
+      if (!active) {
+        return;
+      }
+      setLoaded('loaded');
+    };
+    image.onerror = () => {
+      if (!active) {
+        return;
+      }
+      setLoaded('error');
+    };
+    image.src = src;
+
+    return () => {
+      active = false;
+    };
+  }, [src]);
+
+  return loaded;
+}
+
+const AvatarRoot = styled('div', {
+  name: 'ESAvatar',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const {
+      ownerState: { variant, outlined }
+    } = props;
+
+    return [styles.root, styles[`variant${capitalize(variant)}`], outlined && styles.outlined];
+  }
+})<{ ownerState: AvatarOwnerState & { outlined?: boolean } }>(({ theme, ownerState: { size, outlined } }) => ({
+  height: `${size}px`,
+  width: `${size}px`,
+  backgroundColor: theme.palette.monoA.A100,
+  backgroundSize: '100%',
+  textAlign: 'center',
+  objectFit: 'cover',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  overflow: 'hidden',
+  ...theme.typography.body100,
+  color: theme.palette.monoA.A500,
+
+  ...(outlined && {
+    border: ' 1px solid var(--Light-Mono-A-A100, rgba(0, 0, 0, 0.08))',
```

Comment:
Это внутри `after` перенести, его же делаем спечиально чтобы обводка поверх была. И ещё:
```suggestion
    border: `1px solid ${theme.palette.monoA.A100}`,
```